backend/scripts/mirror_s3_data.sh

```diff
@@ -11,30 +11,109 @@ set -e
+  DEPLOYMENT_STAGE=$SRC_ENV make db/tunnel/up
+  uris=($(PGOPTIONS='-csearch_path=persistence_schema' PGPASSWORD=${DB_PW} psql --dbname corpora_${SRC_ENV} --username corpora_${SRC_ENV} --host 0.0.0.0 --csv --tuples-only -c "$uri_query"))
+  export AWS_PROFILE=single-cell-dev  # For DEST_ENV
+
+  for uri in "${uris[@]}"; do
+    bucket=$(sed -E 's/s3:\/\/([^\/]+).*/\1/' <<< $uri)
+    if [[ -n `grep 'hosted-cellxgene' <<< $bucket` ]]; then
+      rdev_bucket_suffix="cellxgene"
+    elif [[ -n `grep 'corpora-data' <<< $bucket` ]]; then
+      rdev_bucket_suffix="artifacts"
+    fi
+    key_dir=$(sed -E 's/s3:\/\/([^\/]+)\/(.*)/\2/' <<< $uri)
+    sem -j+0 $S3_SYNC_CMD s3://env-rdev-dataportal/s3/${rdev_bucket_suffix}/${sample_data_name}/ s3://env-rdev-${rdev_bucket_suffix}/${STACK}/${key_dir}
+  done
+
+
+  # Copy public assets
+  export AWS_PROFILE=$src_aws_profile  # For SRC_ENV
+  dv_id_query="select distinct unnest(datasets) from \"CollectionVersion\" where collection_id in ('$(sed "s/,/','/g" <<< $COLLECTIONS)')"
+  dataset_version_ids=($(PGOPTIONS='-csearch_path=persistence_schema' PGPASSWORD=${DB_PW} psql --dbname corpora_${SRC_ENV} --username corpora_${SRC_ENV} --host 0.0.0.0 --csv --tuples-only -c "$dv_id_query"))
+  DEPLOYMENT_STAGE=$SRC_ENV make db/tunnel/down  # db access no longer needed
+  export AWS_PROFILE=single-cell-dev  # For DEST_ENV
+
+  exts=("rds" "h5ad")
+  for dv_id in "${dataset_version_ids[@]}"; do
+    for ext in "${exts[@]}"; do
+      uri="s3://dataset-assets-public-${SRC_ENV}/${dv_id}.${ext}"
+      # Only copy if file exists in SRC_ENV public assets bucket (necessary to replicate tombstoned Dataset behavior)
+      sem -j+0 "aws s3 ls $uri && $S3_COPY_CMD s3://env-rdev-dataportal/s3/datasets/${sample_data_name}.${ext} s3://env-rdev-datasets/${STACK}/${dv_id}.${ext}"
+    done
+  done
+  sem --wait
+else  # i.e., if DEST_ENV != 'rdev'
+  # TODO: Add --delete once we confirm that there is no data in the folders
+  # that needs to be kept around (buckets are versioned, so we're not in
+  # jeopardy of losing anything permanently). This would make the
+  # operation more destructive, of course!
+
+  PARALLEL_CMD="parallel --ungroup --jobs 16"  # '--ungroup' permits stdout to stream continuously from all jobs
+
+  hex_chars="0 1 2 3 4 5 6 7 8 9 a b c d e f"  # Permits running 16 concurrent aws processes -- each one syncs only uuids starting with its hex char
+  set -x
+  echo -e "\n\n*** The following parallel aws s3 sync commands may appear to hang, but they are working. Patience is a virtue. ***\n\n"
+  $PARALLEL_CMD $S3_SYNC_CMD s3://corpora-data-${SRC_ENV}/ s3://corpora-data-${DEST_ENV}/ --exclude "'*'" --include "'{}*'" --exclude "'*loom'" ::: $hex_chars
+  $PARALLEL_CMD $S3_SYNC_CMD s3://hosted-cellxgene-${SRC_ENV}/ s3://hosted-cellxgene-${DEST_ENV}/ --exclude "'*'" --include "'{}*'" ::: $hex_chars
```

Comment:
it's really great that github renders all of this ☝️  as though it is a comment block 😭